engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -249,15 +252,10 @@ private void initRenderGraph() {
         renderGraph.add(bloomPassesNode);
         renderGraph.add(blurPassesNode);
         renderGraph.add(finalPostProcessingNode);
-    }
 
-    private static <T extends Node> T createInstance(Class<T> type, Context context) {
-        // Attempt constructor-based injection first
-        T node = InjectionHelper.createWithConstructorInjection(type, context);
-        // Then fill @In fields
-        InjectionHelper.inject(node, context);
-        node.initialise();
-        return type.cast(node);
+        RenderPipelineGenerator generator = new RenderPipelineGenerator(renderGraph);
+        RenderPipelineOptimizer optimizer = new RenderPipelineOptimizer(generator.generate());
+        renderPipelineExecutor = new RenderPipelineExecutor(optimizer.optimize());
```

Comment:
Regarding the above suggestion and the previous comment on the topic, unless we have specific functionality to put into the RenderPipeline and OptimizedPipeline classes I'd convert them both to normal lists. 

I'd then handle them and their transformations within the WorldRenderImpl class, which might eventually become the RenderManager. In particular its preRenderUpdate() method could eventually check for the necessity to refresh the pipelines, while the render() method would do the actual iteration over the tasks in the optimized pipeline. 

By the way, I am now torn regarding how to call the renderPipeline and the optmizedPipeline. Should we call them intermediatePipeline and finalRenderPipeline? Or should we rename the optimized pipeline to "renderTaskList"?

I am suddenly concerned about the term "RenderPipeline" because people could confuse it with an actual rendering pipeline, generating the images. But here the actual rendering pipeline is what we are calling optimized pipeline, which is true but it makes it sound as if the pre-optimized pipeline would also work, it'd just be slower.

As I write this I'm thinking:

```
processableNodes = renderGraph.getNodesInTopologicalOrder();
intermediatePipeline = IntermediatePipelineGenerator.createPipelineFrom(processableNodes);
renderablePipeline = RenderPipelineGenerator.createPipelineFrom(intermediatePipeline);
```